From: local
Subject: [PATCH] centeredwindowname adapted for dwm 6.8 + prior patches

Center the selected window title in the status bar.
Keeps barpadding title width (w - 2 * sp).

--- a/dwm.c
+++ b/dwm.c
@@ -746,8 +746,9 @@
 
 	if ((w = m->ww - tw - x) > bh) {
 		if (m->sel) {
+			int mid = (m->ww - TEXTW(m->sel->name)) / 2 - x;
 			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w - 2 * sp, bh, lrpad / 2, m->sel->name, 0);
+			drw_text(drw, x, 0, w - 2 * sp, bh, mid, m->sel->name, 0);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
